File: Src/Icm.Core.Tests/CommandLineTools/ValuesStoreTest.cs

using Microsoft.VisualBasic;
using System;
using System.Collections;
using System.Collections.Generic;
using System.Data;
using System.Diagnostics;
using Icm.CommandLineTools;

///<summary>
///This is a test class for ValuesStore
///</summary>
[TestFixture(), Category("Icm")]
public class ValuesStoreTest
{

	[TestCase(null, "asdf", ExpectedException = typeof(ArgumentNullException))]
	[TestCase("asdf", null, ExpectedException = typeof(ArgumentNullException))]
	[TestCase("a", "a", ExpectedException = typeof(ArgumentException))]
	[TestCase("a", "asdf")]
	public void AddTest(string shortName, string longName)
	{
		ValuesStore store = new ValuesStore();

		store.AddParameter(shortName, longName);

		Assert.That(store.ContainsKey(shortName));
		Assert.That(store.ContainsKey(longName));
	}

	[TestCase(null, "asdf", ExpectedException = typeof(ArgumentNullException))]
	[TestCase("asdf", null, ExpectedException = typeof(ArgumentNullException))]
	[TestCase("a", "a", ExpectedException = typeof(ArgumentException))]
	[TestCase("asdf", "a", ExpectedException = typeof(ArgumentException))]
	[TestCase("asdf", "qwer", ExpectedException = typeof(ArgumentException))]
	[TestCase("a", "asdf")]
	public void AddValue(string shortName, string longName)
	{
		ValuesStore store = new ValuesStore();

		store.AddParameter(shortName, longName);

		Assert.That(store.ContainsKey(shortName));
		Assert.That(store.ContainsKey(longName));
	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================

File: Src2/Icm.Core.Tests/CommandLineTools/ValuesStoreTest.vb
Imports Icm.CommandLineTools

'''<summary>
'''This is a test class for ValuesStore
'''</summary>
<TestFixture(), Category("Icm")>
Public Class ValuesStoreTest

    <TestCase(Nothing, "asdf", ExpectedException:=GetType(ArgumentNullException))>
    <TestCase("asdf", Nothing, ExpectedException:=GetType(ArgumentNullException))>
    <TestCase("a", "a", ExpectedException:=GetType(ArgumentException))>
    <TestCase("a", "asdf")>
    Public Sub AddTest(shortName As String, longName As String)
        Dim store As New ValuesStore

        store.AddParameter(shortName, longName)

        Assert.That(store.ContainsKey(shortName))
        Assert.That(store.ContainsKey(longName))
    End Sub

    <TestCase(Nothing, "asdf", ExpectedException:=GetType(ArgumentNullException))>
    <TestCase("asdf", Nothing, ExpectedException:=GetType(ArgumentNullException))>
    <TestCase("a", "a", ExpectedException:=GetType(ArgumentException))>
    <TestCase("asdf", "a", ExpectedException:=GetType(ArgumentException))>
    <TestCase("asdf", "qwer", ExpectedException:=GetType(ArgumentException))>
    <TestCase("a", "asdf")>
    Public Sub AddValue(shortName As String, longName As String)
        Dim store As New ValuesStore

        store.AddParameter(shortName, longName)

        Assert.That(store.ContainsKey(shortName))
        Assert.That(store.ContainsKey(longName))
    End Sub

End Class
